File: iosApp/iosApp/views/resources/text/TextResources.swift
import Foundation

func stringRes(_ key: String, _ args: CVarArg...) -> String {
    let format = NSLocalizedString(key, comment: "")
    return String(format: format, arguments: args)
}

struct Strings {
    static var homeTitle: String { stringRes("screen_home_title") }
    static var favoritesTitle: String { stringRes("screen_favorites_title") }
    static var searchTitle: String { stringRes("screen_search_title") }
    static var settingsTitle: String { stringRes("screen_settings_title") }

    static var navBrowse: String { stringRes("nav_browse") }
    static var navFavorites: String { stringRes("nav_favorites") }

    static var appLoadingMessage: String { stringRes("app_loading_message") }
    static var appStartupErrorTitle: String { stringRes("app_startup_error_title") }
    static var appStartupErrorSyncDesc: String { stringRes("app_startup_error_sync_desc") }
    static var appStartupErrorDesc: String { stringRes("app_startup_error_desc") }
    static var appErrorTryAgain: String { stringRes("app_error_try_again") }

    static var searchPlaceholder: String { stringRes("search_placeholder") }
    static var noSearchResult: String { stringRes("no_search_result") }
    static var startSearch: String { stringRes("start_search") }
    static var detailsCoatOfArms: String { stringRes("details_coat_of_arms") }
    static var commonClose: String { stringRes("common_close") }
    static var commonConfirm: String { stringRes("common_confirm") }
    static var commonContinue: String { stringRes("common_continue") }
    static var commonCancel: String { stringRes("common_cancel") }

    static var detailLabelLocation: String { stringRes("detail_label_location") }
    static var detailLabelArea: String { stringRes("detail_label_area") }
    static var detailLabelPopulation: String { stringRes("detail_label_population") }
    static var detailLabelLanguages: String { stringRes("detail_label_languages") }
    static var detailLabelCurrency: String { stringRes("detail_label_currency") }
    static var detailLabelTimezones: String { stringRes("detail_label_timezones") }
    static var detailLabelCapital: String { stringRes("detail_label_capital") }
    static var detailLabelRegion: String { stringRes("detail_label_region") }

    static var listItemLabelCapital: String { stringRes("list_item_label_capital") }
    static func listItemLabelCapital(_ capital: String) -> String {
        return stringRes("list_item_label_capital_fmt", capital)
    }

    static var commonView: String { stringRes("common_view") }
    static var commonRemove: String { stringRes("common_remove") }
    static var commonBack: String { stringRes("common_back") }
    static var commonClear: String { stringRes("common_clear") }
    static var commonSettings: String { stringRes("common_settings") }
    static var commonMore: String { stringRes("common_more") }
    static var commonMoreOptions: String { stringRes("common_more_options") }
    static var commonSearch: String { stringRes("common_search") }
    static var commonOpenInMaps: String { stringRes("common_open_in_maps") }
    static var commonOpenInWikipedia: String { stringRes("common_open_in_wikipedia") }
    static var openKeyboard: String { stringRes("open_keyboard") }


    static var emptyTitleNoResults: String { stringRes("empty_title_no_results") }
    static var emptyMsgNoResults: String { stringRes("empty_msg_no_results") }
    static var emptyTitleEmptyList: String { stringRes("empty_title_empty_list") }
    static var emptyMsgEmptyList: String { stringRes("empty_msg_empty_list") }
    static var emptyTitleNotFound: String { stringRes("empty_title_not_found") }
    static var emptyMsgNotFound: String { stringRes("empty_msg_not_found") }

    static func detailLabelLanguage(count: Int) -> String {
        return count == 1 ? stringRes("detail_label_language") : stringRes("detail_label_languages")
    }
}

File: composeApp/src/androidMain/kotlin/com/exploramus/app/resources/TextResources.kt
package com.exploramus.app.resources

import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import com.exploramus.app.R

@Composable
fun stringRes(
    resId: Int,
    vararg formatArgs: Any
): String {
    return stringResource(resId, *formatArgs)
}

object Strings {
    val homeTitle @Composable get() = stringRes(R.string.screen_home_title)
    val favoritesTitle @Composable get() = stringRes(R.string.screen_favorites_title)
    val searchTitle @Composable get() = stringRes(R.string.screen_search_title)
    val settingsTitle @Composable get() = stringRes(R.string.screen_settings_title)

    val navBrowse @Composable get() = stringRes(R.string.nav_browse)
    val navFavorites @Composable get() = stringRes(R.string.nav_favorites)

    val appStartupErrorTitle @Composable get() = stringRes(R.string.app_startup_error_title)
    val appStartupErrorSyncDesc @Composable get() = stringRes(R.string.app_startup_error_sync_desc)
    val appStartupErrorDesc @Composable get() = stringRes(R.string.app_startup_error_desc)
    val appErrorTryAgain @Composable get() = stringRes(R.string.app_error_try_again)

    val searchPlaceholder @Composable get() = stringRes(R.string.search_placeholder)
    val noSearchResult @Composable get() = stringRes(R.string.no_search_result)
    val startSearch @Composable get() = stringRes(R.string.start_search)
    val detailsCoatOfArms @Composable get() = stringRes(R.string.details_coat_of_arms)
    val commonClose @Composable get() = stringRes(R.string.common_close)
    val commonConfirm @Composable get() = stringRes(R.string.common_confirm)
    val commonCancel @Composable get() = stringRes(R.string.common_cancel)

    val detailLabelLocation @Composable get() = stringRes(R.string.detail_label_location)
    val detailLabelArea @Composable get() = stringRes(R.string.detail_label_area)
    val detailLabelPopulation @Composable get() = stringRes(R.string.detail_label_population)
    val detailLabelLanguages @Composable get() = stringRes(R.string.detail_label_languages)
    val detailLabelCurrency @Composable get() = stringRes(R.string.detail_label_currency)
    val detailLabelTimezones @Composable get() = stringRes(R.string.detail_label_timezones)
    val detailLabelCapital @Composable get() = stringRes(R.string.detail_label_capital)
    val detailLabelRegion @Composable get() = stringRes(R.string.detail_label_region)

    val listItemLabelCapital @Composable get() = stringRes(R.string.list_item_label_capital)
    @Composable fun listItemLabelCapital(capital: String) = stringRes(R.string.list_item_label_capital_fmt, capital)

    val commonView @Composable get() = stringRes(R.string.common_view)
    val commonRemove @Composable get() = stringRes(R.string.common_remove)
    val commonBack @Composable get() = stringRes(R.string.common_back)
    val commonClear @Composable get() = stringRes(R.string.common_clear)
    val commonPrevious @Composable get() = stringRes(R.string.common_previous)
    val commonNext @Composable get() = stringRes(R.string.common_next)
    val commonSearch @Composable get() = stringRes(R.string.common_search)
    val commonSettings @Composable get() = stringRes(R.string.common_settings)
    val commonMore @Composable get() = stringRes(R.string.common_more)
    val commonMoreOptions @Composable get() = stringRes(R.string.common_more_options)
    val commonOpenInMaps @Composable get() = stringRes(R.string.common_open_in_maps)
    val commonOpenInWikipedia @Composable get() = stringRes(R.string.common_open_in_wikipedia)
    val openKeyboard @Composable get() = stringRes(R.string.open_keyboard)
    val appLoadingMessage @Composable get() = stringRes(R.string.app_loading_message)

    val emptyTitleNoResults @Composable get() = stringRes(R.string.empty_title_no_results)
    val emptyMsgNoResults @Composable get() = stringRes(R.string.empty_msg_no_results)
    val emptyTitleEmptyList @Composable get() = stringRes(R.string.empty_title_empty_list)
    val emptyMsgEmptyList @Composable get() = stringRes(R.string.empty_msg_empty_list)
    val emptyTitleNotFound @Composable get() = stringRes(R.string.empty_title_not_found)
    val emptyMsgNotFound @Composable get() = stringRes(R.string.empty_msg_not_found)

    val flashcardSettingsTitle @Composable get() = stringRes(R.string.flashcard_settings_title)
    val flashcardVisibleClue @Composable get() = stringRes(R.string.flashcard_visible_clue)
    val flashcardTargetPrimary @Composable get() = stringRes(R.string.flashcard_target_primary)
    val flashcardTargetSecondary @Composable get() = stringRes(R.string.flashcard_target_secondary)
    val flashcardTargetImage @Composable get() = stringRes(R.string.flashcard_target_image)
    val flashcardSettingsDescription @Composable get() = stringRes(R.string.flashcard_settings_description)
    val flashcardShuffleCards @Composable get() = stringRes(R.string.flashcard_shuffle_cards)
    val flashcardRevealDetails @Composable get() = stringRes(R.string.flashcard_reveal_details)
    val flashcardRestart @Composable get() = stringRes(R.string.flashcard_restart)
    val flashcardDone @Composable get() = stringRes(R.string.flashcard_done)
    val flashcardHintTapToReveal @Composable get() = stringRes(R.string.flashcard_hint_tap_to_reveal)
    val flashcardMenuSettings @Composable get() = stringRes(R.string.flashcard_menu_settings)
    val flashcardLabelCapital @Composable get() = stringRes(R.string.flashcard_label_capital)

    val quizTargetPrimarySecondary @Composable get() = stringRes(R.string.quiz_target_primary_secondary)
    val quizTargetSecondaryPrimary @Composable get() = stringRes(R.string.quiz_target_secondary_primary)
    val quizTargetImagePrimary @Composable get() = stringRes(R.string.quiz_target_image_primary)
    val quizTargetPrimaryImage @Composable get() = stringRes(R.string.quiz_target_primary_image)

    val choiceQuizSettingsTitle @Composable get() = stringRes(R.string.choice_quiz_settings_title)
    val choiceQuizSelectPair @Composable get() = stringRes(R.string.choice_quiz_select_pair)
    val choiceQuizLimitLabel @Composable get() = stringRes(R.string.choice_quiz_limit_label)
    val choiceQuizLimitNoLimit @Composable get() = stringRes(R.string.choice_quiz_limit_no_limit)

    @Composable
    fun detailLabelLanguage(count: Int) = if (count == 1)
        stringRes(R.string.detail_label_language)
    else
        stringRes(R.string.detail_label_languages)
}








